### ISSUE TITLE
[Disussion] GitHub Actions for labeling Pull Requests

### ISSUE THREAD

**Issue by Sc0rpi0n101** [sympy/sympy#18221]
I think it might be a good idea to set up an Github actions workflow to label Pull reuests according to the modules they affect. It might save a lot of time for the mentainers who have to manually label a lot of Pull Requests. 

We can also set up a workflow for stalled Pull Requests as per #17683 

CC: @asmeurer @czgdp1807

**Comment by czgdp1807** [comment]
It looks like a nice idea. I will test this on one of my active projects. If everything works out fine and members of the community support this we will setup the labeller with sympy.
Though there will be still a need to check the correctness of the label as rules cannot be generalized to be correct in every situation though I am sure it will reduce the burden a lot. Thanks for sharing this.

**Comment by Sc0rpi0n101** [comment]
Great, I'm also learning and working with Actions for another project. So, if you get a lot of work and need help with this, I'd be happy to help.

> Though there will be still a need to check the correctness of the label as rules cannot be generalized to be correct in every situation though I am sure it will reduce the burden a lot. Thanks for sharing this.

Yeah, we will have to keep a close look at the labels, especially for the stalled PRs. With over 500 PRs ranging till 2012, some of them are bound to be wrong. As for generalization, we can use the Policy on [Closing PRs](https://github.com/sympy/sympy/wiki/Closing-pull-requests) as a base for the workflow. But, we'd need to complete the discussion in #17683 and update the policy before starting with the workflow.

For other labels, we can just set up the workflow to label the correct module as a start. The rest of the labels would still need the maintainers to apply them manually.

**Comment by asmeurer** [comment]
We could also have the bot do this. The hard part is getting the logic which labels to apply correct.

**Comment by czgdp1807** [comment]
Yes, in fact, when I label a PR I use a different logic for different situations. Author's turn label cannot be left to a bot because a general rule cannot work there as it's a subjective concept. If we think carefully, there at most 4-5 labels for a PR in the worst case, now whether done by a bot or by a human doesn't make a difference as at the end it's human who will check. 
Sometimes, stalled PRs become active once the author is reminded of their work and authors do that with dedication. So, no general rule can define which PR is stalled and when, as again humans verify that finally.
I formed the above views after thinking about this issue for some time. Comments from the community are required to decide on this.

**Comment by oscarbenjamin** [comment]
I'm not sure that automating the labelling of PRs is really worthwhile. It doesn't take long to label a PR compared with the time taken to read or review a PR. We don't have any problem right now with remembering to add labels: all new PRs and issues get labels.

**Comment by sylee957** [comment]
I think that for pull requests, it can be fine for bot to entirely depend on the release notes to judge.
For issues, it can be difficult unless we give such a survey template.

**Comment by czgdp1807** [comment]
For PRs the following are the scenarios where the bot can fail,

1. `NO ENTRY` release notes where files are required to be checked manually and labels are to be attached. Github Actions can automatically check the files and attach labels but sometimes irrelevant files are also changed which don't require a label to be used for them. 
2. `Please take over` PRs are to be manually checked for their status and `Please take over` PRs is attached after manual judgement. Similar is the case with `Could Close` PRs.
3.  Number of new PRs is not much, at most 10 per day, which can be labelled manually along side providing reviews.
